core/cli/src/config/hash.ts

```diff
@@ -17,20 +18,26 @@ export async function fileHash(path: string): Promise<string> {
   }
 }
 
-export async function updateHashes(): Promise<void> {
+export async function updateHashes(config: ValidConfig): Promise<void> {
   const hashes = Object.fromEntries(
-    await Promise.all(configPaths.map(async (path) => [path, await fileHash(path)]))
+    await Promise.all(
+      ['.toolkitrc.yml', ...config.hookManagedFiles].map(async (path) => [path, await fileHash(path)])
+    )
   )
   writeState('install', hashes)
 }
 
-export async function hasConfigChanged(logger: Logger): Promise<boolean> {
+export async function hasConfigChanged(logger: Logger, config: ValidConfig): Promise<boolean> {
   const hashes = readState('install')
+
   if (!hashes) {
     return true
   }
-  for (const [path, prevHash] of Object.entries(hashes)) {
+
+  for (const path of ['.toolkitrc.yml', ...config.hookManagedFiles]) {
```

Comment:
**suggestion(non-blocking)**: think we could also consider the config to have changed if `Object.keys(hashes) !== ['.toolkitrc.yml', ...config.hookManagedFiles]` to catch cases where a hook has been removed. though idk if that would ever require a reinstall?

Reply:
it's impossible for a _removed_ hook to cause a difference with install so i don't think we need this